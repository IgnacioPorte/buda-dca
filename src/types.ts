type HistoricalPrice = {
  price: number;
  date: string;
};

type MonthlyDCAInvestment = {
  price: number;
  date: string;
  fiatInvestment: number;
  cryptoPortfolio: number;
  portfolioFiatValue: number;
  gain: number;
  gainPercentage: number;
};


export type { HistoricalPrice, MonthlyDCAInvestment };