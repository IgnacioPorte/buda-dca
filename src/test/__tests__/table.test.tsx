import Table from "@/components/dcaTable";
import { test, expect, describe } from "vitest";
import { MonthlyDCAInvestment } from "@/types";
import { render } from "@testing-library/react";

const mockData: MonthlyDCAInvestment[] = [
  {
    cryptoPortfolio: 0.1,
    date: "2021-01",
    fiatInvestment: 100,
    gain: 0,
    gainPercentage: 0,
    portfolioFiatValue: 150,
    price: 150,
  },
  {
    cryptoPortfolio: 0.2,
    date: "2021-02",
    fiatInvestment: 200,
    gain: 50,
    gainPercentage: 25,
    portfolioFiatValue: 250,
    price: 250,
  },
];

describe("Table Component Tests", () => {
  test("renders without crashing", () => {
    expect(render(<Table data={[]} />)).toMatchSnapshot();
  });

  test("renders chart with data", () => {
    const table = render(<Table data={mockData} />);
    expect(table).toMatchSnapshot();
    expect(table.getByText("2021-01")).toBeDefined();
    expect(table.getByText("25.00%")).toBeDefined();
  });
});
