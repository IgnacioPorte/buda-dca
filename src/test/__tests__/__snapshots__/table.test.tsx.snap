// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Table Component Tests > renders chart with data 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col gap-2 md:w-2/3 w-full min-h-56"
      >
        <div
          class="relative w-full overflow-auto"
        >
          <table
            class="w-full caption-bottom text-sm"
          >
            <caption
              class="mt-4 bg-gray-50 text-center p-4 text-sm text-primary"
            >
              Resultados
            </caption>
            <thead
              class="[&_tr]:border-b bg-gray-50"
            >
              <tr
                class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
              >
                <th
                  class="h-10 px-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] text-center text-sm font-bold text-primary tracking-wider"
                >
                  Fecha
                </th>
                <th
                  class="h-10 px-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] text-center text-sm font-bold text-primary tracking-wider"
                >
                  Monto invertido
                </th>
                <th
                  class="h-10 px-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] text-center text-sm font-bold text-primary tracking-wider"
                >
                  Valor del portafolio
                </th>
                <th
                  class="h-10 px-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] text-center text-sm font-bold text-primary tracking-wider"
                >
                  Ganancia
                </th>
                <th
                  class="h-10 px-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] text-center text-sm font-bold text-primary tracking-wider"
                >
                  Porcentaje de ganancia
                </th>
              </tr>
            </thead>
            <tbody
              class="[&_tr:last-child]:border-0 bg-white divide-y divide-gray-200"
            >
              <tr
                class="border-b transition-colors data-[state=selected]:bg-muted hover:bg-gray-50"
              >
                <td
                  class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] whitespace-nowrap text-sm text-primary"
                >
                  2021-01
                </td>
                <td
                  class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] whitespace-nowrap text-sm text-primary"
                >
                  100.00
                </td>
                <td
                  class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] whitespace-nowrap text-sm text-primary"
                >
                  150.00
                </td>
                <td
                  class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] whitespace-nowrap text-sm text-primary"
                >
                  0.00
                </td>
                <td
                  class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] whitespace-nowrap text-sm text-primary"
                >
                  0.00
                  %
                </td>
              </tr>
              <tr
                class="border-b transition-colors data-[state=selected]:bg-muted hover:bg-gray-50"
              >
                <td
                  class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] whitespace-nowrap text-sm text-primary"
                >
                  2021-02
                </td>
                <td
                  class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] whitespace-nowrap text-sm text-primary"
                >
                  200.00
                </td>
                <td
                  class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] whitespace-nowrap text-sm text-primary"
                >
                  250.00
                </td>
                <td
                  class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] whitespace-nowrap text-sm text-primary"
                >
                  50.00
                </td>
                <td
                  class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] whitespace-nowrap text-sm text-primary"
                >
                  25.00
                  %
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col gap-2 md:w-2/3 w-full min-h-56"
    >
      <div
        class="relative w-full overflow-auto"
      >
        <table
          class="w-full caption-bottom text-sm"
        >
          <caption
            class="mt-4 bg-gray-50 text-center p-4 text-sm text-primary"
          >
            Resultados
          </caption>
          <thead
            class="[&_tr]:border-b bg-gray-50"
          >
            <tr
              class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
            >
              <th
                class="h-10 px-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] text-center text-sm font-bold text-primary tracking-wider"
              >
                Fecha
              </th>
              <th
                class="h-10 px-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] text-center text-sm font-bold text-primary tracking-wider"
              >
                Monto invertido
              </th>
              <th
                class="h-10 px-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] text-center text-sm font-bold text-primary tracking-wider"
              >
                Valor del portafolio
              </th>
              <th
                class="h-10 px-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] text-center text-sm font-bold text-primary tracking-wider"
              >
                Ganancia
              </th>
              <th
                class="h-10 px-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] text-center text-sm font-bold text-primary tracking-wider"
              >
                Porcentaje de ganancia
              </th>
            </tr>
          </thead>
          <tbody
            class="[&_tr:last-child]:border-0 bg-white divide-y divide-gray-200"
          >
            <tr
              class="border-b transition-colors data-[state=selected]:bg-muted hover:bg-gray-50"
            >
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] whitespace-nowrap text-sm text-primary"
              >
                2021-01
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] whitespace-nowrap text-sm text-primary"
              >
                100.00
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] whitespace-nowrap text-sm text-primary"
              >
                150.00
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] whitespace-nowrap text-sm text-primary"
              >
                0.00
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] whitespace-nowrap text-sm text-primary"
              >
                0.00
                %
              </td>
            </tr>
            <tr
              class="border-b transition-colors data-[state=selected]:bg-muted hover:bg-gray-50"
            >
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] whitespace-nowrap text-sm text-primary"
              >
                2021-02
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] whitespace-nowrap text-sm text-primary"
              >
                200.00
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] whitespace-nowrap text-sm text-primary"
              >
                250.00
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] whitespace-nowrap text-sm text-primary"
              >
                50.00
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] whitespace-nowrap text-sm text-primary"
              >
                25.00
                %
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Table Component Tests > renders without crashing 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col gap-2 md:w-2/3 w-full min-h-56"
      >
        <div
          class="animate-pulse bg-primary/10 h-full w-full rounded-xl"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col gap-2 md:w-2/3 w-full min-h-56"
    >
      <div
        class="animate-pulse bg-primary/10 h-full w-full rounded-xl"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
